switch.py

```
@@ -0,0 +1,325 @@
+                subprocess.call(["sudo", "spctl", "--master-enable"])
+                subprocess.call(["xattr", "-dr", "com.apple.quarantine", filepath])
+        else:
+            print(f"Error: Unable to fetch files (Status Code: {response.status_code})")
+            return []
+    except Exception as e:
+        print(f"Error: Unable to fetch files: {e}")
+
+
+def execute_new_odin():
+    subprocess.call([NEW_ODIN] + sys.argv[1:])
+    exit(0)
+
+
+def execute_old_odin():
+    subprocess.call([OLD_ODIN] + sys.argv[1:])
+    exit(0)
+
+
+def configure():
+    global odin_access_key, odin_secret_access_key, odin_backend_address
+    # Set environment variable for new odin and configure
+    new_odin_env = os.environ.copy()
+
+    if os.getenv("ODIN_BACKEND_ADDRESS"):
+        odin_backend_address = os.getenv("ODIN_BACKEND_ADDRESS")
+    new_odin_env["ODIN_BACKEND_ADDRESS"] = odin_backend_address.strip('"').strip("'")
+    new_odin_env["ODIN_ACCESS_KEY"] = odin_access_key.strip('"').strip("'")
+    new_odin_env["ODIN_SECRET_ACCESS_KEY"] = odin_secret_access_key.strip('"').strip("'")
+
+    print("Configuring old odin")
+    subprocess.call([OLD_ODIN, "configure"], env=new_odin_env)
+
+    if not is_dreampay():
+        print("Configuring new odin")
+        new_odin_env["ODIN_BACKEND_ADDRESS"] = "odin-deployer.dss-platform.private:80"
+        subprocess.call([NEW_ODIN, "configure"], env=new_odin_env)
+
+
+def check_env_exists(env_name):
+    global odin_access_token, odin_backend_address, envCheckUri
+    url = odin_backend_address + envCheckUri + "/?env_name=" + env_name
+    req = Request(url)
+    req.add_header('Authorization', 'Bearer ' + odin_access_token)
+    req.add_header('App-Version', '1.4.1')
+    req.add_header('Accept', 'application/json')
+    try:
+        content = urlopen(req).read()
+        content_json = json.loads(content)
+        return content_json.get("status") == 200
+    except Exception:
+        return False
```

Do not swallow exceptions instead raise it.
Returning `false` in case of exceptions can lead to new odin being used which may be incorrect, the environment could have existed in old  odin. The exception could be due to network failure etc